filter.go

```diff
@@ -0,0 +1,48 @@
+package files
+
+import (
+	"path/filepath"
+
+	ignore "github.com/crackcomm/go-gitignore"
+)
+
+// Filter - Filter filter.
+type Filter struct {
+	// IncludeHidden - Include hidden files
+	IncludeHidden bool
+	// Rules - File filter rules
+	Rules *ignore.GitIgnore
+}
+
+// NewFilter creates a new file filter form .gitignore and list of rules.
+func NewFilter(ignoreFile string, rules []string, includeHidden bool) (filter *Filter, err error) {
+	filter = &Filter{IncludeHidden: includeHidden}
+	if ignoreFile == "" {
+		filter.Rules, err = ignore.CompileIgnoreLines(rules...)
+	} else {
+		filter.Rules, err = ignore.CompileIgnoreFileAndLines(ignoreFile, rules...)
+	}
+	return
+}
+
+func isPathHidden(path string) bool {
```

Comment:
Can we assume that a file starting with a dot character is hidden on all platforms (eg windows)?

Reply:
See https://github.com/ipfs/go-ipfs-files/blob/6d2d295fc352776b91fa076319dcfc54ca1d7147/is_hidden_windows.go

We should consolidate all the hidden file detection code in one place.

Reply:
See my response to your other [suggestion](https://github.com/ipfs/go-ipfs-files/pull/26#discussion_r384613162)